blog_post_collaboration/kedro_ibis.md

```diff
@@ -0,0 +1,263 @@
+# conf/base/catalog_connections.yml
+_duckdb:
+  backend: duckdb
+  # `database` and `threads` are parameters for `ibis.duckdb.connect()`.
+  database: jaffle_shop.duckdb
+  threads: 24
+
+# conf/base/catalog.yml
+raw_customers:
+  type: jaffle_shop.datasets.ibis.TableDataset
+  table_name: raw_customers
+  connection: ${_duckdb}
+  save_args:
+    materialized: table
+
+...
+```
+
+### Building pipelines
+
+Develop pipelines the same way you would any other Kedro pipeline. Below, we show a node written using Ibis alongside the original dbt model. Note that Ibis solves the [parametrisation problem](https://www.youtube.com/watch?v=XdZklxTbCEA&t=514s) with vanilla Python syntax; on the other hand, the dbt counterpart uses Jinja templating.
+
+<table>
+ <tr>
+  <th>Kedro node
+  <th>dbt model
+ <tr valign="top">
+  <td>
+
+```python
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import ibis
+from ibis import _
+
+if TYPE_CHECKING:
+    import ibis.expr.types as ir
+
+
+def process_orders(
+    orders: ir.Table, payments: ir.Table, payment_methods: list[str]
+) -> ir.Table:
+    total_amount_by_payment_method = {}
+    for payment_method in payment_methods:
+        total_amount_by_payment_method[f"{payment_method}_amount"] = ibis.coalesce(
+            payments.amount.sum(where=payments.payment_method == payment_method), 0
+        )
+
+    order_payments = payments.group_by("order_id").aggregate(
+        **total_amount_by_payment_method, total_amount=payments.amount.sum()
+    )
+
+    final = orders.left_join(order_payments, "order_id")[
+        orders.order_id,
+        orders.customer_id,
+        orders.order_date,
+        orders.status,
+        *[
+            order_payments[f"{payment_method}_amount"]
+            for payment_method in payment_methods
+        ],
+        order_payments.total_amount.name("amount"),
+    ]
+    return final
+```
+
+  <td>
+
+```sql
+{% set payment_methods = ['credit_card', 'coupon', 'bank_transfer', 'gift_card'] %}
+
+with orders as (
+
+    select * from {{ ref('stg_orders') }}
+
+),
+
+payments as (
+
+    select * from {{ ref('stg_payments') }}
+
+),
+
+order_payments as (
+
+    select
+        order_id,
+
+        {% for payment_method in payment_methods -%}
+        sum(case when payment_method = '{{ payment_method }}' then amount else 0 end) as {{ payment_method }}_amount,
+        {% endfor -%}
+
+        sum(amount) as total_amount
+
+    from payments
+
+    group by order_id
+
+),
+
+final as (
+
+    select
+        orders.order_id,
+        orders.customer_id,
+        orders.order_date,
+        orders.status,
+
+        {% for payment_method in payment_methods -%}
+
+        order_payments.{{ payment_method }}_amount,
+
+        {% endfor -%}
+
+        order_payments.total_amount as amount
+
+    from orders
+
+
+    left join order_payments
+        on orders.order_id = order_payments.order_id
+
+)
+
+select * from final
+```
+
+</table>
+
+### Quickstart
```

Comment:
Weird to see "Quickstart" almost at the end of the blog post. Maybe "See it for yourself"?